File: samples/ble_util/src/com/idevicesinc/sweetblue/ble_util/ViewController.java
package com.idevicesinc.sweetblue.ble_util;

import com.idevicesinc.sweetblue.BleManager;

import android.content.Context;
import android.view.View;
import android.widget.FrameLayout;
import android.widget.LinearLayout;

/**
 * 
 * @author dougkoellmer
 */
public class ViewController extends LinearLayout
{
	private static enum State
	{
		DEVICE_LIST,
		DEVICE_DETAIL;
	}
	
	private final BleManager m_bleMngr;
	private final FrameLayout m_inner;
	
	private State m_state = null; 
	
	public ViewController(Context context, BleManager bleMngr)
	{
		super(context);
		
		m_bleMngr = bleMngr;
		
		this.setLayoutParams(new LayoutParams(LayoutParams.MATCH_PARENT, LayoutParams.MATCH_PARENT));
		this.setOrientation(VERTICAL);
		
		BleBar bleStateBar = new BleBar(context, bleMngr);
		bleStateBar.setLayoutParams(new LinearLayout.LayoutParams(LayoutParams.MATCH_PARENT, LayoutParams.WRAP_CONTENT));
		bleStateBar.setBackgroundColor(0x33A21615);
		this.addView(bleStateBar);
		
		m_inner = new FrameLayout(context);
		m_inner.setLayoutParams(new LinearLayout.LayoutParams(LayoutParams.MATCH_PARENT, LayoutParams.MATCH_PARENT));
		this.addView(m_inner);
		
		setState(State.DEVICE_LIST);
	}
	
	private void setState(State state)
	{
		if( m_state != null )
		{
			m_inner.removeAllViews();
		}
		
		m_state = state;
		
		View newView = null;
		
		if( m_state == State.DEVICE_LIST )
		{
			newView = new DeviceList(getContext(), m_bleMngr);
		}
		else if( m_state == State.DEVICE_DETAIL )
		{
		}
		
		if( newView != null )
		{
			newView.setLayoutParams(new LayoutParams(LayoutParams.MATCH_PARENT, LayoutParams.MATCH_PARENT));
			m_inner.addView(newView);
		}
	}
}

File: xamarin/SweetBlue/component/samples/Android/BleUtil/BleUtil/ViewController.cs
using System;
using Android.Views;
using Android.Widget;
using Android.Content;
using Idevices.Sweetblue;


namespace BleUtil
{
	public class ViewController : LinearLayout
	{

		private enum State {
			DEVICE_LIST,
			DEVICE_DETAIL,
			NULL
		}

		private BleManager m_bleMgr;
		private FrameLayout m_inner;
		private State m_state = State.NULL;


		public ViewController (Context context, BleManager mgr) : base(context)
		{
			m_bleMgr = mgr;

			this.LayoutParameters = new LayoutParams (LayoutParams.MatchParent, LayoutParams.MatchParent);
			this.Orientation = Orientation.Vertical;

			BleBar bleStateBar = new BleBar (context, mgr);
			bleStateBar.LayoutParameters = new LayoutParams (LayoutParams.MatchParent, LayoutParams.WrapContent);
			bleStateBar.SetBackgroundColor (new Android.Graphics.Color(0x33A21615));
			AddView (bleStateBar);

			m_inner = new FrameLayout (context);
			m_inner.LayoutParameters = new LayoutParams (LayoutParams.MatchParent, LayoutParams.MatchParent);
			AddView (m_inner);

			setState (State.DEVICE_LIST);
		}

		private void setState(State state) {
			if (m_state != State.NULL) {
				m_inner.RemoveAllViews ();
			}

			m_state = state;

			View newView = null;

			if (m_state == State.DEVICE_LIST) {
				newView = new DeviceList (Context, m_bleMgr);
			} else if (m_state == State.DEVICE_DETAIL) {				
			}

			if (newView != null) {
				newView.LayoutParameters = new LayoutParams (LayoutParams.MatchParent, LayoutParams.MatchParent);
				m_inner.AddView (newView);
			}				
		}

	}
}

